common/network_manager.go

```diff
@@ -347,37 +347,52 @@ func (m *NetworkManager) onLoadingFailed(event *network.EventLoadingFailed) {
 }
 
 func (m *NetworkManager) onLoadingFinished(event *network.EventLoadingFinished) {
-	req := m.requestFromID(event.RequestID)
+	rid := event.RequestID
+	req := m.requestForOnLoadingFinished(rid)
+
+	// the request was not created yet.
 	if req == nil {
-		// Handling of iframe document request starting in parent session and ending up in iframe session.
-		if m.parent != nil {
-			reqFromParent := m.parent.requestFromID(event.RequestID)
-
-			// Main requests have matching loaderID and requestID.
-			if reqFromParent != nil && reqFromParent.getDocumentID() == event.RequestID.String() {
-				m.reqsMu.Lock()
-				m.reqIDToRequest[event.RequestID] = reqFromParent
-				m.reqsMu.Unlock()
-				m.parent.deleteRequestByID(event.RequestID)
-				req = reqFromParent
-			} else {
-				return
-			}
-		} else {
-			return
-		}
+		return
 	}
+
 	req.responseEndTiming = float64(event.Timestamp.Time().Unix()-req.timestamp.Unix()) * 1000
 	// Skip data and blob URLs when emitting metrics, since they're internal to the browser.
 	if !isInternalURL(req.url) {
 		req.responseMu.RLock()
 		m.emitResponseMetrics(req.response, req)
 		req.responseMu.RUnlock()
 	}
-	m.deleteRequestByID(event.RequestID)
+	m.deleteRequestByID(rid)
 	m.frameManager.requestFinished(req)
 }
 
+// requestForOnLoadingFinished returns the request for the given request ID.
+func (m *NetworkManager) requestForOnLoadingFinished(rid network.RequestID) *Request {
+	r := m.requestFromID(rid)
+
+	// Immediately return if the request is not found.
+	if r != nil {
+		return r
+	}
+
+	// Handle IFrame document requests starting in one session and ending up in another.
+	if m.parent == nil {
+		return nil
+	}
+	// Main requests have matching loaderID and requestID.
```

Comment:
What's a "Main" request and "loaderID"? I don't see it being referenced anywhere else.

Reply:
Yep, that's the existing, old comment. It is supposed to mean requests from the main frame. A common CDP term, `LoaderID` refers to the frame that loads the request.

Reply:
Right, got it. Could you update the comment to make it a bit clearer please? 🙂

Reply:
[8d8e3e4](https://github.com/grafana/xk6-browser/pull/1084/commits/8d8e3e46ad130bc09f9243f8770f300ea2c2dfc8)

Reply:
Btw, I've also noticed there's another bug 😅